jgrss/crs error

jgrss opened the pull request:
## What is this PR changing?
This PR fixes a bug with unsupported EPSG codes in calls to `to_raster()`. For example, [EPSG:8858](https://epsg.io/8858) cannot be translated to WKT directly by `rasterio.CRS`. Instead, it requires a translation to `pyproj.CRS`, which can then be used to translate to a string by `to_wkt()` or `to_proj4()`.

jgrss commented on setup.cfg:
```diff
@@ -34,7 +34,7 @@ install_requires =
     pandas>=1.*             # via -r requirements.in, dask, geopandas, xarray
     pyproj>=2.6.0             # via -r requirements.in, geopandas
-    rasterio>=1.2.0,<=1.2.6
+    rasterio>=1.3.0,<2.0.0
```
This supports `stackstac`.

jgrss commented on setup.cfg:
```diff
@@ -76,7 +76,7 @@ view = ipython
     requests
     pystac-client
-    stackstac[viz]
+    stackstac[viz]@git+https://github.com/jgrss/stackstac.git  # contains some pyproj temp fixes
```
I had to fork and make some minor `pyproj` changes.

jgrss commented on src/geowombat/core/io.py:
```diff
@@ -769,16 +770,27 @@ def to_raster(
         if isinstance(kwargs['num_threads'], str):
             kwargs['num_threads'] = 'all_cpus'
 
+    if 'crs' in kwargs:
```
This is the main change. I.e., translating input CRS codes to `pyproj.CRS` classes that can support any EPSG code.

jgrss commented on src/geowombat/radiometry/sr.py:
```diff
@@ -95,29 +95,36 @@ def s_atm(r):
-                          '6VCID2': 'th2', '7': 'swir2', '8': 'pan'},
-                      l8={'1': 'coastal', '2': 'blue', '3': 'green', '4': 'red', '5': 'nir', '6': 'swir1',
-                          '7': 'swir2', '8': 'pan', '9': 'cirrus', '10': 'th1', '11': 'th2'})
+    bands_dict = dict(
```
This is already merged and didn't get squashed.

jgrss commented on src/geowombat/radiometry/sr.py:
```diff
@@ -148,12 +150,17 @@ def get_landsat_coefficients(meta_file):
-                        'LANDSAT_8': 'l8'}
-
-        MetaCoeffs = namedtuple('MetaCoeffs', 'sensor m_l a_l m_p a_p date_acquired sza left bottom right top')
+        associations = {
```
Also already merged

mmann1123 reviewed (approved):
Looks good to me. Although we need to start adding tests as we go. Sorry I've been distracted with other work.

jgrss commented:
> Looks good to me. Although we need to start adding tests as we go. Sorry I've been distracted with other work.

Yeah, good call. I added a few things to this PR, mainly a change from `rasterio.CRS` to `pyproj.CRS` to better support EPSG codes. I also updated a couple of tests.
